src/main/java/com/adobe/aem/compgenerator/AemCompGenerator.java

```diff
@@ -40,44 +72,151 @@
+                    port = Integer.parseInt(portArr[1]);
+                } catch (NumberFormatException e) {
+                    System.out.println("Invalid arg usage, expected -p=8080");
+                    System.exit(1);
+                }
+            } else {
+                // check if user wants to just run the component generator from an existing json config
+                if (StringUtils.contains(argument, ".json")) {
+                    File configFile = new File(argument);
+                    if (CommonUtils.isFileBlank(configFile)) {
+                        throw new GeneratorException("Config file missing / empty.");
+                    }
+                    GenerationConfig config = CommonUtils.getComponentData(configFile);
+                    if (config == null) {
+                        throw new GeneratorException("Config file is empty / null !!");
+                    }
 
-            if (CommonUtils.isFileBlank(configFile)) {
-                throw new GeneratorException("Config file missing / empty.");
-            }
+                    if (!config.isValid() || !CommonUtils.isModelValid(config.getProjectSettings())) {
+                        throw new GeneratorException("Mandatory fields missing in the data-config.json !");
+                    }
 
-            GenerationConfig config = CommonUtils.getComponentData(configFile);
+                    String compDir = config.getProjectSettings().getAppsPath() + "/"
+                            + config.getProjectSettings().getComponentPath() + "/"
+                            + config.getType() + "/" + config.getName();
+                    config.setCompDir(compDir);
 
-            if (config == null) {
-                throw new GeneratorException("Config file is empty / null !!");
+                    //builds component folder and file structure.
+                    try {
+                        ComponentUtils generatorUtils = new ComponentUtils(config);
+                        generatorUtils.buildComponent();
+
+                        //builds sling model based on config.
+                        if (config.getOptions() != null && config.getOptions().isHasSlingModel()) {
+                            JavaCodeModel javaCodeModel = new JavaCodeModel();
+                            javaCodeModel.buildSlingModel(config);
+                        }
+                        System.out.println("Your component has been generated.");
+                        System.exit(0);
+                    } catch (Exception e) {
+                        LOG.error("Failed to generate aem component.", e);
+                    }
+                } else {
+                    System.out.println("Invalid arg usage, expected -p=8080, or pass a valid path to a config json file");
+                }
             }
+        }
+        // Ensure that an initial data configuration
+        // JSON file exists for loading / saving data to
+        String configPath = "data-config.json";
```

Comment:
`main()` has become 200 lines long - let's modularize (which will also help bring clarity to what all this code is doing)